### PR TITLE
apps sc: harbor np image replication

### LINKED ISSUE #1308: [1] Harbor networkpolicy blocking replication
**Describe the bug**
The harbor networkpolicies do not have any config options to allow communication with other container registries, e.g. for replicating images. At least core and registry should have an option to allow egress traffic to other registries, but testing is needed to see if other components also need access. They currently have config for object storage, but I do not think that is intended to be used for this.

**To Reproduce**
1. Enable network policies for harbor
2. Try to set up replication from another registry
3. See error about timeout

**Expected behavior**
There is config to allow traffic to other registries (list of IPs and ports).

**Screenshots**

**Version (add all relevant versions):**
 - Compliant kubernetes apps version v0.26.3

**Additional context**

### PR DESCRIPTION
**Pull request by robinAwallace**
**What this PR does / why we need it**:

Only need to update core to fix replication.
Separated harbor nps

**Which issue this PR fixes**: fixes #1308 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [x] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [x] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        - ipBlock:
+            cidr: {{ $ip }}
+        {{- end }}
+      {{- else if .Values.global.ingressUsingHostNetwork }}
+        {{- range $ip := .Values.global.scNodes.ips }}
+        - ipBlock:
+            cidr: {{ $ip }}
+        {{- end }}
+      {{- else }}
+        - namespaceSelector:
```
Comment (Xartos) [inline helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml:31]: Don't you also need to add [this](https://github.com/elastisys/compliantkubernetes-apps/pull/1349/files#diff-09c655a89fde669d4a28279e488666530397122a163f85710998d3f0d184f78dR56-R59)
```suggestion
        {{- range $IP := .Values.global.ingress.ips }}
        - ipBlock:
            cidr: {{ $IP }}
        {{- end }}
        - namespaceSelector:
```

Comment (robinAwallace) [reply helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml:31]: Ah yes, that should still be there. Will fix that

**Comment by viktor-f** [comment]
Did you test chart replication as well?

File: helmfile/charts/networkpolicy-service/values.yaml [hunk 2]
```diff
@@ -47,6 +47,9 @@ global:
 
 harbor:
   enabled: true
+  imageReplication:
+    ips:
+      - "0.0.0.0/0"
```
Comment (viktor-f) [inline helmfile/charts/networkpolicy-service/values.yaml:52]: Is this used for anything?

**Comment by robinAwallace** [comment]
> Did you test chart replication as well?

Yes it works, to and from harbor
image

**Comment by robinAwallace** [comment]
I tried cache proxy and it works.

File: helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml [hunk 3]
```diff
@@ -162,6 +162,7 @@ spec:
             component: jobservice
       ports:
         - port: 8080
+    {{- if and .Values.harbor.registries.ips .Values.harbor.registries.ports }}
```
Comment (robinAwallace) [inline helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml:165]: @viktor-f 
This is my solution for this bug
https://github.com/elastisys/compliantkubernetes-apps/issues/1363

Comment (viktor-f) [reply helmfile/charts/networkpolicy-service/templates/harbor/allow-harbor-core.yaml:165]: Looks good

**Review by viktor-f — APPROVED** [review]
LGTM, just squash at lest some of your commits

**Review by Xartos — APPROVED** [review]

**Pull request merged**